Add doc about removing Stripe and payments (#316)
sharetribe/flex-docs: src/docs/cookbook-payments/removing-stripe/index.md

---
title: How to remove Stripe and payments
slug: removing-stripe-and-payments
updated: 2020-07-06
category: cookbook-payments
ingress:
  In some cases, you might want to remove Stripe integrations or even all the
  payments from FTW. This article gives you a starting point for these
  customizations.
published: true
---

# Removing Stripe

In this guide, we will go through the minimum changes needed to make the booking

> flow work without Stripe. Most likely you need to customize some parts (e.g.

Reviewer: I think this should include a bit more about what to expect. This guide is very rough by nature - clients clearly need to go through lots of (unguided) changes on several files. So, this paragraph could prepare devs & admins that going through these changes is going to be a relatively big customization project.

 FTW templates are made for a specific transaction process.
- Payments are a core feature that we have tried to cover with these provided templates.

```suggestion
flow work without Stripe. This is not an easy customization project. FTW templates are made for a specific transaction process and payments are a core feature: its ubiquitous nature means that you need to touch quite many components and files. Most likely you need to customize some parts (e.g.
```

transaction process, email templates, checkout page) more depending on your

> marketplace idea. You might also want to remove e.g. unused Stripe components

Reviewer: ```suggestion
marketplace idea. For example, you might also want to remove unused Stripe components
```

Reviewer: Or, "E.g. [...]"

from your project to clean up the code.

###1. Update the transaction process and remove Stripe related actions

You need to remove at least the following actions:

- :action/stripe-create-payment-intent
- :action/stripe-capture-payment-intent
- :action/stripe-confirm-payment-intent
- :action/stripe-create-payout
- :action/stripe-refund-charge
- :action/stripe-refund-payment

If you are using the FTW default process with strong customer authentication
(e.g. flex-default-process) you need to remove confirming the payment.
Otherwise, the transaction will get stuck. The simplest way to do this is to
remove `pending-payment` and `expire-payment` states and point `request-payment`

> and `equest-payment-after-enquiry` directly to `preauthorized` state. However,

Reviewer: ```suggestion
and `request-payment-after-enquiry` directly to `preauthorized` state. However,
```

when you are building the transaction process to your marketplace it might make

> sense to rename some of the states and transitions (e.g. preauthorized) so that
> they make more sense in your use-case.

Reviewer: ```suggestion
sense to rename some of the states and transitions so that
they make more sense in your use-case. E.g. *preauthorized* state refers to the state of payment - it is preauthorized to be captured.
```

Example:

```diff
     [STATE_INITIAL]: {
       on: {
         [TRANSITION_ENQUIRE]: STATE_ENQUIRY,
-        [TRANSITION_REQUEST_PAYMENT]: STATE_PENDING_PAYMENT,
+        [TRANSITION_REQUEST_PAYMENT]: STATE_PREAUTHORIZED,
       },
     },
     [STATE_ENQUIRY]: {
       on: {
-        [TRANSITION_REQUEST_PAYMENT_AFTER_ENQUIRY]: STATE_PENDING_PAYMENT,
+        [TRANSITION_REQUEST_PAYMENT_AFTER_ENQUIRY]: STATE_PREAUTHORIZED,
       },
     },
-
-    [STATE_PENDING_PAYMENT]: {
-      on: {
-        [TRANSITION_EXPIRE_PAYMENT]: STATE_PAYMENT_EXPIRED,
-        [TRANSITION_CONFIRM_PAYMENT]: STATE_PREAUTHORIZED,
-      },
-    },
-
-    [STATE_PAYMENT_EXPIRED]: {},
     [STATE_PREAUTHORIZED]: {
       on: {
         [TRANSITION_DECLINE]: STATE_DECLINED,

```

###2. Remove Stripe checks from EditListingWizard

In
[`EditListingWizard.js`](https://github.com/sharetribe/ftw-daily/blob/master/src/components/EditListingWizard/EditListingWizard.js)
we are checking if the provider has a Stripe account with all the required
information before we allow them to publish listings. This check is done in the
`handlePublishListing` function and needs to be removed.

```js
  handlePublishListing(id) {

>     const { onPublishListingDraft} = this.props;
>     onPublishListingDraft(id);

Reviewer: ```suggestion
    this.props.onPublishListingDraft(id);
```

And actually, this whole function could be removed, but that would require another line to be edited (the one where this.handlePublishListing(id) is called. For the sake of this guide, it's probably better to keep the change inside this one method.

  }
```

After changing the `handlePublishListing` you can remove unused code. You also

> might want to clean up `EditListingPage` and remove the Stripe related props we

Reviewer: Is there any link to the correct lines (where to begin this cleaning) or is there too many of them?

Author: On `EditLisitngPage` there are mostly props that are passed forward and some variables related to Stripe so not very easy to point to just one line.

pass to `EditListingWizard`.

###3. Edit ModalMissingInformation

By default,
[`ModalMissingInformation`](https://github.com/sharetribe/ftw-daily/blob/master/src/components/ModalMissingInformation/ModalMissingInformation.js)
will remind users to create a Stripe account. The same modal is used for
reminding about email verification so we should just remove the Stripe related
code from the component.

###4. Edit CheckoutPage

Removing Stripe from `CheckoutPage` is probably the most complicated task in
this list because the Strong Customer Authentication (SCA) and ability to save
payment method have added a lot of logic to the current page. We can use the

> [edited version of CheckoutPage from FTW 2.17.1](/tutorial-assets/SimpleCheckoutPage.js)

Reviewer: Don't hide a new resource as a link inside paragraph. It's easier to spot these if they are more prominent.

----

We can use the edited version of CheckoutPage from FTW 2.17.1 as a simpler starting point:

- [SimpleCheckoutPage.js](/tutorial-assets/SimpleCheckoutPage.js)

as a simpler starting point.

Some of the functions on later versions of CheckoutPage.duck.js has changed so
here are the main changes done in the edited code:

- In `handleSubmit` function we only call onInitiateOrder function which
  internally decides if the transaction should be initiated or transitioned. We

>   also need to handle sending the possible inital message after that.

Reviewer: ```suggestion
  also need to handle sending the possible initial message after that.
```

- StripeConnectAccountForm

You can also remove Stripe script form
[public/index.html](https://github.com/sharetribe/ftw-daily/blob/master/public/index.html#L169).

# Removing payments

If you want to remove payments completely from your marketplace, there are even
more things to consider. Again it depends a lot on your use-case what kind of
changes are needed. If you e.g. want to add the second transaction process with
free transactions you need to build the conditional logic for showing pricing
components when needed and hiding them when not. If you want to build the
marketplace completely without payments then you most likely want to hide all
the references to the listing price.

### More changes to the transaction process

Remove the actions related to payments e.g. in flex-default-process
:action/privileged-set-line-items and :action/calculate-full-refund. In older
processes this means actions related to calculating total price, commission etc.
Read more about actions from
[transaction process actions refrence article.](/references/transaction-process-actions/)
Remember also to update email templates which contain pricing information.

### Edit API calls in FTW backend

> If you are using FTW-daily v6.0.0 or higher or FTW-hourly v8.0.0 or higher you

Reviewer: Lot's of "or" conjunctions that are connecting items a bit strangely. 

One alternative suggestion:

```suggestion
If you are using FTW-daily v6.0.0 or higher (or FTW-hourly v8.0.0 or higher) you
```
